data-plane/core/src/test/java/dev/knative/eventing/kafka/broker/core/reconciler/impl/EgressReconcilerListenerMock.java

```diff
@@ -0,0 +1,63 @@
+package dev.knative.eventing.kafka.broker.core.reconciler.impl;
+
+import dev.knative.eventing.kafka.broker.contract.DataPlaneContract;
+import dev.knative.eventing.kafka.broker.core.reconciler.EgressReconcilerListener;
+import io.vertx.core.Future;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+public class EgressReconcilerListenerMock implements EgressReconcilerListener {
+
+  private final Set<String> newEgresses;
+  private final Set<String> updatedEgresses;
+  private final Set<String> deletedEgresses;
```

Comment:
I'd prefer a structure that keeps track of duplicates without hiding duplicate entries.